src/libs/antares/study/parts/short-term-storage/properties.h

```diff
@@ -0,0 +1,77 @@
+/*
+** Copyright 2007-2023 RTE
+** Authors: Antares_Simulator Team
+**
+** This file is part of Antares_Simulator.
+**
+** Antares_Simulator is free software: you can redistribute it and/or modify
+** it under the terms of the GNU General Public License as published by
+** the Free Software Foundation, either version 3 of the License, or
+** (at your option) any later version.
+**
+** There are special exceptions to the terms and conditions of the
+** license as they are applied to this software. View the full text of
+** the exceptions in file COPYING.txt in the directory of this software
+** distribution
+**
+** Antares_Simulator is distributed in the hope that it will be useful,
+** but WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+** GNU General Public License for more details.
+**
+** You should have received a copy of the GNU General Public License
+** along with Antares_Simulator. If not, see <http://www.gnu.org/licenses/>.
+**
+** SPDX-License-Identifier: licenceRef-GPL3_WITH_RTE-Exceptions
+*/
+
+#include <optional>
+#pragma once
+
+#include <string>
+#include <map>
+
+#include <antares/inifile.h>
+
+namespace Antares::Data::ShortTermStorage
+{
+enum class Group
+{
+    PSP_open,
+    PSP_closed,
+    Pondage,
+    Battery,
+    Other1,
+    Other2,
+    Other3,
+    Other4,
+    Other5
+};
+
+unsigned int groupIndex(Group group);
+
+class Properties
+{
+public:
+    bool validate();
+    bool loadKey(const IniFile::Property* p);
```

Comment:
Yes, I am not saying that there is an error in the current code, but the point is to make clear to the caller of this method, who could possibly be anyone in the future since it's public, that he must not pass a null argument.

Using a reference is a semantic way in c++ to document that an argument is not nullable.